app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -85,6 +83,54 @@ private String[] getAvailableMoves(String[] board) {
         }
 
         return Arrays.copyOf(availableMovesArray.toArray(), availableMovesArray.size(), String[].class);
+//        return Arrays.stream(board).filter((space) -> isSpaceAvailable(board.index)).toArray(String[]::new);
     }
 
+    private String[] mapBoardMovesToSet(int[] set) {
+        String[] setBeingChecked = new String[set.length];
+
+        for (int i = 0; i < set.length; i++) {
+            setBeingChecked[i] = board[set[i]];
+        }
+
+        return setBeingChecked;
+    }
+
+    private int[][] createWinningSet(int boardSize) {
+        int[][] winningSets = new int[(boardSize * 2) + 2][boardSize];
```

Comment:
Consider starting with a list of zero winners that you append to as you go along, rather than a list of null values that you replace as you go along.